tools/firtool/firtool.cpp

```diff
@@ -372,6 +377,10 @@ static LogicalResult processBuffer(
   if (failed(firtool::populatePreprocessTransforms(pm, firtoolOptions)))
     return failure();
 
+  if (enableDebugInfo)
+    pm.nest<firrtl::CircuitOp>().addNestedPass<firrtl::FModuleOp>(
+        firrtl::createMaterializeDebugInfoPass());
```

Comment:
nit: should we move this to the end of the firtool::populatePreprocessTransforms? We are trying to provide these libraries for users looking to extend firtool downstream, and having loose passes in the top-level of firtool.cpp means they need to duplicate the firtool.cpp logic in their downstream. Maybe it makes sense for this specific pass, but in general I think it's a reasonable goal to have no passmanager mutation directly in firtool.cpp.

Reply:
Yeah that's a good point. I'm going to move the pass into the library.